server/src/main/java/org/opensearch/search/profile/query/InternalProfileCollectorManager.java

```diff
@@ -51,7 +52,9 @@ public InternalProfileCollectorManager(
 
     @Override
     public InternalProfileCollector newCollector() throws IOException {
-        return new InternalProfileCollector(manager.newCollector(), reason, children);
+        InternalProfileCollector profileCollector = new InternalProfileCollector(manager.newCollector(), reason, children);
+        collectorName = profileCollector.getName();
```

Comment:
It does not look right: manager may create different collectors, we should not rely on when generating the name of the collector manager.

Reply:
Hi @reta, I understand your point, and it has been addressed. Please take another look, thanks.